File: remove-bg.js
const Jimp = require('jimp');

async function removeWhiteBackground(inputPath, outputPath) {
  try {
    const image = await Jimp.read(inputPath);
    
    // We iterate through all pixels
    image.scan(0, 0, image.bitmap.width, image.bitmap.height, function(x, y, idx) {
      // Get RGBA values
      const red = this.bitmap.data[idx + 0];
      const green = this.bitmap.data[idx + 1];
      const blue = this.bitmap.data[idx + 2];
      
      // If the pixel is very close to white (allow some tolerance for compression artifacts)
      if (red > 240 && green > 240 && blue > 240) {
        // Set alpha to 0 (transparent)
        this.bitmap.data[idx + 3] = 0;
      }
    });

    await image.writeAsync(outputPath);
    console.log('Background removed successfully.');
  } catch (error) {
    console.error('Error removing background:', error);
  }
}

removeWhiteBackground('public/logo-icon.png', 'public/logo-icon-transparent.png');

File: remove-bg.py
from PIL import Image

def remove_white_bg(input_path, output_path):
    img = Image.open(input_path).convert("RGBA")
    datas = img.getdata()
    
    newData = []
    for item in datas:
        # white threshold
        if item[0] > 240 and item[1] > 240 and item[2] > 240:
            newData.append((255, 255, 255, 0)) # transparent
        else:
            newData.append(item)
            
    img.putdata(newData)
    img.save(output_path, "PNG")

remove_white_bg("public/logo-icon.png", "public/logo-icon-transparent.png")
